broadinstitute/single_cell_portal_core
# Allow clustering updates in AnnData file upload (SCP-5039)

Allow multiple clusterings, updating and deleting clusterings for an AnnData file upload


To test:
- Pull this branch and start up delayed jobs and a rails server
- Ensure you have the flag for AnnData ingest flipped to true
- Start from a fresh study (necessary as there was a bug in the way clusterings were handled previously)
- Fill in expression and clustering info
- For clustering start with one and make the name "first" and the obsm key "X_umap"
- upload the AnnData file 
- allow this to complete uploading and parsing

To test updating a name
- Now go to the clustering page and change the name of the clustering and click save
- Observe in `development.log` that an ingest job will be kicked off to re-parse this
- Observe in the Study details page the clustering will be deleted and then replaced with the new name

To test adding another clustering
- go to the clustering page and click to add a clustering
- name it "second" and give it an obsm key name of "X_tsne" 
- click save
- now wait, locally this takes a bit of time for me, but eventually you will see the job in `development.log` and refreshing the study details page a few times you will see this new clustering option available.

You can also delete a clustering (it won't present a delete button if there is only one though) and you will see this reflected in the study details page as well.

Here is a screenrecording of me going through all this - warning it's 16min haha this is not quick on local https://drive.google.com/file/d/1egNdyYy081KW5xHPdlsBAsFn3XWOOzUk/view?usp=sharing
